### PR TITLE
[EASY REVIEW] Edit checkbox table

### PR DESCRIPTION
**Pull request by kennyt**

### CODE REVIEW

File: src/js/components/CheckboxTable.js [hunk 1]
```diff
@@ -48,12 +48,13 @@ class CheckboxTable extends React.Component {
 
   bulkCheck(isChecked) {
     let checkedIDs = [];
-    let {data, onCheckboxChange} = this.props;
+    let {data, onCheckboxChange, uniqueProperty} = this.props;
 
     if (isChecked) {
       data.forEach(function (datum) {
-        checkedIDs.push(datum.id);
+        checkedIDs.push(datum[uniqueProperty]);
```
Comment (kennyt) [inline src/js/components/CheckboxTable.js]: we don't always want to look at the `id` field. we should choose what the id is

File: src/js/components/CheckboxTable.js [hunk 2]
```diff
@@ -104,7 +106,7 @@ class CheckboxTable extends React.Component {
   renderCheckbox(prop, row) {
     let {checkedItemsMap, uniqueProperty} = this.props;
     let rowID = row[uniqueProperty];
-    let checked = null;
+    let checked = false;
```
Comment (kennyt) [inline src/js/components/CheckboxTable.js]: form doesn't work if it's null. needs to be false

File: src/js/components/CheckboxTable.js [hunk 3]
```diff
@@ -164,7 +166,7 @@ class CheckboxTable extends React.Component {
         buildRowOptions={this.getTableRowOptions}
         className={tableClassSet}
         columns={columns}
-        colGroup={this.getColGroup()}
+        colGroup={getColGroup()}
```
Comment (kennyt) [inline src/js/components/CheckboxTable.js]: allow external col group

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/components/CheckboxTable.js (hunk 3 above)
Comment (jfurrow) [inline src/js/components/CheckboxTable.js]: Can we just add this to `propTypes`? And maybe add a noop as the `defaultProp` as I don't think we'll need it.

**Comment by jfurrow** [comment]
After my one comment, it looks good!

**Pull request merged**